src/backend/avx2/edwards.rs

```diff
@@ -0,0 +1,1057 @@
+
+            // Set t0 = (X1 Y1 Z1 X1+Y1)
+            // why does this intrinsic take an i32 for the imm8 ???
+            t0.0[0] = _mm256_blend_epi32(P.0[0].into(), t0.0[0].into(), D_LANES as i32).into();
+            t0.0[1] = _mm256_blend_epi32(P.0[1].into(), t0.0[1].into(), D_LANES as i32).into();
+            t0.0[2] = _mm256_blend_epi32(P.0[2].into(), t0.0[2].into(), D_LANES as i32).into();
+            t0.0[3] = _mm256_blend_epi32(P.0[3].into(), t0.0[3].into(), D_LANES as i32).into();
+            t0.0[4] = _mm256_blend_epi32(P.0[4].into(), t0.0[4].into(), D_LANES as i32).into();
+
+            // Set t1 = t0^2, negating the D values
+            t1 = t0.square(D_LANES64);
+
+            // Now t1 = (S1 S2 S3 -S4)
+
+            let c0 = u32x8::new(0,0,2,2,0,0,2,2); // (ABCD) -> (AAAA)
+            let c1 = u32x8::new(1,1,3,3,1,1,3,3); // (ABCD) -> (BBBB)
+
+            // See discussion of bounds in the module-level documentation.
+            //
+            // We want to compute
+            // 
+            //    + | S1 | S1 | S1 | S1 |  
+            //    + | S2 |    |    | S2 |  
+            //    + |    |    | S3 |    |  
+            //    + |    |    | S3 |    |  
+            //    + |    |    |    |-S4 |
+            //    + |    | 2p | 2p |    |  
+            //    - |    | S2 | S2 |    |  
+            //    =======================  
+            //        S5   S6   S8   S9    
+            //
+            for i in 0..5 {
+                let zero = i32x8::splat(0);
+                let S1 = _mm256_permutevar8x32_epi32(t1.0[i], c0);
+                let S2 = _mm256_permutevar8x32_epi32(t1.0[i], c1);
+                let S3_2: u32x8 = _mm256_blend_epi32(zero, (t1.0[i] + t1.0[i]).into(), 0b01010000).into();
+                // tmp0 = (0 0 2*S3 -S4)
+                let tmp0: u32x8 = _mm256_blend_epi32(S3_2.into(), t1.0[i].into(), 0b10100000).into();
+                t0.0[i] = (avx2::constants::P_TIMES_2_MASKED.0[i] + tmp0) + S1;
+                t0.0[i] = t0.0[i] + _mm256_blend_epi32(zero, S2.into(), 0b10100101).into();
+                t0.0[i] = t0.0[i] - _mm256_blend_epi32(S2.into(), zero, 0b10100101).into();
+            }
+
+            let c0 = u32x8::new(4,0,6,2,4,0,6,2); // (ABCD) -> (CACA)
+            let c1 = u32x8::new(5,1,7,3,1,5,3,7); // (ABCD) -> (DBBD)
+
+            for i in 0..5 {
+                let tmp = t0.0[i];
+                t0.0[i] = _mm256_permutevar8x32_epi32(tmp, c0);
+                t1.0[i] = _mm256_permutevar8x32_epi32(tmp, c1);
+            }
+
+            ExtendedPoint(&t0 * &t1)
+        }
+    }
+
+    pub fn mult_by_pow_2(&self, k: u32) -> ExtendedPoint {
+        let mut tmp: ExtendedPoint = *self;
+        for _ in 0..k {
+            tmp = tmp.double();
+        }
+        tmp
+    }
+}
+
+impl<'a, 'b> Add<&'b CachedPoint> for &'a ExtendedPoint {
+    type Output = ExtendedPoint;
+
+    /// Uses a slight tweak of the parallel unified formulas of HWCD'08
```

Comment:
Maybe document what was tweaked more explicitly?

Reply:
I can add a note that the tweaks are documented in the module-level docs.